File: Caelan.Frameworks.Common.NUnit/DateTimeRangeTest.cs
using System;
using Caelan.Frameworks.Common.Classes;
using NUnit.Framework;

namespace Caelan.Frameworks.Common.NUnit
{
    [TestFixture]
    public class DateTimeRangeTest
    {
        [Test]
        public void TestRange()
        {
            var range = new DateTimeRange(DateTime.Today, DateTime.Now);

            Assert.IsTrue(range.IsInRange(DateTime.Now.AddHours(-1)));
        }
    }
}

File: Caelan.Frameworks.Common.FS.NUnit/DateTimeRangeTest.fs
namespace Caelan.Frameworks.Common.NUnit

open System
open System.Diagnostics
open NUnit.Framework
open Caelan.Frameworks.Common.Classes

[<TestFixture>]
type DateTimeRangeTest() = 
    [<Test>]
    member __.TestRange() = 
        let range = DateTimeRange(DateTime.Today, DateTime.Now)
        Assert.IsTrue(range.IsInRange(DateTime.Now.AddHours(float (-1))))
